test/bench/codec.bench.ts

```diff
@@ -0,0 +1,46 @@
+import { itBench } from '@dapplion/benchmark'
+import { Flag, FrameHeader, FrameType } from '../../src/frame.js'
+import { encodeFrame } from '../../src/encode.js'
+import { decodeHeader } from '../../src/decode.js'
+import { decodeHeaderNaive, encodeFrameNaive } from '../codec.util.js'
+
+describe('codec benchmark', () => {
+  for (const { encode, name } of [
+    { encode: encodeFrame, name: 'encodeFrame' },
+    { encode: encodeFrameNaive, name: 'encodeFrameNaive' }
+  ]) {
+    itBench<FrameHeader, undefined>({
+      id: `encode a frame header - ${name}`,
```

Comment:
For easier lookup latter it's best to have benchmark names in the shortest form, ordered from most general to least:
```
`frame header ${name}`
`frame header ${name}`
```